src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -199,7 +202,7 @@ private void UpdateCheckoutTasksAndVariables(IExecutionContext executionContext,
                 }
 
                 // Update the checkout "Clean" property for all repos, if the variable was set by the server.
-                if (repoClean != null)
+                if (repoClean != null && repoCleanFromServer)
                 {
                     checkoutTask.Inputs[PipelineConstants.CheckoutTaskInputs.Clean] = repoClean.Value.ToString();
```

Comment:
Should there be assigned a value which comes from server only?

Reply:
Yes, it assigning only values that come from the server, since this block is placed before the block where repoClean is overwriting based on the input of Self checkout task, it setting task input according to server value and due to it, the value of repoClean won't change [here](https://github.com/microsoft/azure-pipelines-agent/blob/9b232182ad3f262edf6e0c786baaedaef3b6106a/src/Agent.Worker/Build/BuildJobExtension.cs#L220).